test/unit/custom_spec.js

```diff
@@ -107,3 +107,91 @@ describe("custom canvas rendering", function () {
       .catch(done.fail);
   });
 });
+
+describe("alternate document context", function () {
+  const FontFace = global.FontFace;
+
+  let CanvasFactory;
+
+  beforeAll(() => {
+    global.FontFace = class {};
+    CanvasFactory = isNodeJS ? new NodeCanvasFactory() : new DOMCanvasFactory();
+  });
+
+  afterAll(() => {
+    global.FontFace = FontFace;
+  });
+
+  it("should use given document for loading fonts", async function () {
+    const domDoc = { fonts: new Set() };
+    const getDocumentParams = buildGetDocumentParams(
+      "TrueType_without_cmap.pdf",
+      {
+        disableFontFace: false,
+        ownerDocument: domDoc,
+      }
+    );
+
+    const doc = await getDocument(getDocumentParams).promise;
+
+    const page = await doc.getPage(1);
+
+    const viewport = page.getViewport({ scale: 1 });
+    const canvasAndCtx = CanvasFactory.create(viewport.width, viewport.height);
+
+    await page.render({
+      canvasContext: canvasAndCtx.context,
+      viewport,
+    }).promise;
+
+    expect(domDoc.fonts.size).toBe(1);
+
+    await doc.destroy();
+
+    expect(domDoc.fonts.size).toBe(0);
+  });
+
+  it("should use given document for loading fonts (legacy path)", async function () {
+    const sheet = {
+      cssRules: [],
+      insertRule(rule) {
+        this.cssRules.push(rule);
+      },
+    };
+    const getDocumentParams = buildGetDocumentParams(
+      "TrueType_without_cmap.pdf",
+      {
+        disableFontFace: false,
+        ownerDocument: {
+          fonts: null,
+          createElement(x) {
```

Comment:
As a rule we don't want to use one-letter variable/argument names, since those are generally too non-descriptive.

Reply:
Yeah, was going for a minimum viable mock for the interface. I've updated,